### PR TITLE
AbsSys Abund

### PR DESCRIPTION
**Pull request by profxj**
A series of edits related to measuring abundances in 
an AbsSystem.

Tests, docs, notebook, etc.

### CODE REVIEW

File: linetools/line_utils.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            return np.array(out_list)
+    else:
+        return out_list
+
+
+def transtable_from_speclines(abslines, add_keys=None):
+    """Generate a Table summarizing the transitions from a list of AbsLines
+    Parameters
+    ----------
+    speclines : list of SpectralLine objects
```
Comment (ntejos) [inline linetools/line_utils.py]: speclines -> abslines

File: linetools/line_utils.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+from .spectralline import AbsLine
+
+def parse_speclines(abslines, key, mk_array=False):
+    """ Generate a list or array of items from a list of AbsLines
+
+    Parameters
+    ----------
+    abslines : list of AbsLine objects
+    key : str
```
Comment (ntejos) [inline linetools/line_utils.py]: Please add a better description for `key` parameter.

File: linetools/line_utils.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
```
Comment (ntejos) [inline linetools/line_utils.py]: Please add description of `mk_array` parameter.

File: linetools/line_utils.py (hunk 1 above)
Comment (ntejos) [inline linetools/line_utils.py]: Please add description for `add_keys`

File: linetools/isgm/abssystem.py [hunk 4]
```diff
@@ -351,6 +360,66 @@ def list_of_abslines(self):
...
+        abs_lines = self.list_of_abslines()
+        # Loop
+        for iline in abs_lines:
+            # Fill in spec?
+            if spec is not None:
+                iline.analy['spec'] = spec
+            # Measure
+            iline.measure_aodm(**kwargs)
+        #
+        print("You may now wish to update the component column densities")
```
Comment (ntejos) [inline linetools/isgm/abssystem.py]: "You may now wish to update the component column densities. If so, please use `update_component_colm()`"

**Comment by ntejos** [comment]
Left some comments, mostly regarding docstrings. Everything else seemed fine to me.

File: linetools/line_utils.py (hunk 1 above)
Comment (profxj) [inline linetools/line_utils.py]: Changed to speclines (not abslines).

Add doc

File: linetools/line_utils.py (hunk 2 above)
Comment (profxj) [inline linetools/line_utils.py]: Done

File: linetools/line_utils.py (hunk 3 above)
Comment (profxj) [inline linetools/line_utils.py]: Done.

File: linetools/isgm/abssystem.py (hunk 4 above)
Comment (profxj) [inline linetools/isgm/abssystem.py]: Done

**Comment by profxj** [comment]
Merging

**Pull request merged**